packages/whale-api-client/__tests__/api/governance.test.ts

```diff
@@ -0,0 +1,375 @@
+    const noVote = result.find(vote => vote.vote === ProposalVoteResultType.NO)
+    const neutralVote = result.find(vote => vote.vote === ProposalVoteResultType.NEUTRAL)
+    expect(result.length).toStrictEqual(3)
+    expect(yesVote).toStrictEqual({
+      proposalId: cfpProposalId,
+      masternodeId: expect.any(String),
+      cycle: 2,
+      vote: ProposalVoteResultType.YES
+    })
+    expect(noVote).toStrictEqual({
+      proposalId: cfpProposalId,
+      masternodeId: expect.any(String),
+      cycle: 2,
+      vote: ProposalVoteResultType.NO
+    })
+    expect(neutralVote).toStrictEqual({
+      proposalId: cfpProposalId,
+      masternodeId: expect.any(String),
+      cycle: 2,
+      vote: ProposalVoteResultType.NEUTRAL
+    })
+  })
+
+  it('should listProposalVotes with cycle', async () => {
+    const result = await client.governance.listGovProposalVotes({ id: cfpProposalId, cycle: 2 })
+    expect(result.length).toStrictEqual(3)
+  })
+
+  it('should listProposalVotes with all records when all flag is true', async () => {
+    const result = await client.governance.listGovProposalVotes({ id: cfpProposalId, all: true })
+    expect(result.length).toStrictEqual(3)
+  })
+
+  it('should listProposalVotes with all masternodes', async () => {
+    const result = await client.governance.listGovProposalVotes({ id: cfpProposalId, masternode: MasternodeType.ALL })
+    expect(result.length).toStrictEqual(3)
+  })
+
+  it('should listProposalVotes with all masternodes and cycle', async () => {
+    const result = await client.governance.listGovProposalVotes({ id: cfpProposalId, masternode: MasternodeType.ALL, cycle: -1 })
+    expect(result.length).toStrictEqual(4)
+
+    const result2 = await client.governance.listGovProposalVotes({ id: cfpProposalId, masternode: MasternodeType.ALL, cycle: 0 })
+    expect(result2.length).toStrictEqual(3)
+  })
+
+  it('should listProposalVotes with all masternodes, cycle and pagination', async () => {
+    const resultPage1 = await client.governance.listGovProposalVotes({ id: cfpProposalId, masternode: MasternodeType.ALL, cycle: 2, size: 2 })
+    expect(resultPage1.length).toStrictEqual(2)
+    const resultPage2 = await client.governance.listGovProposalVotes({ id: cfpProposalId, masternode: MasternodeType.ALL, cycle: 2, size: 2, next: resultPage1.nextToken })
+    expect(resultPage2.length).toStrictEqual(1)
+  })
+
+  it('should listProposalVotes with error when using non-existent proposal ID', async () => {
+    try {
+      await client.governance.listGovProposalVotes({ id: '123' })
+    } catch (err) {
+      expect(err).toBeInstanceOf(WhaleApiException)
+      expect(err.error).toStrictEqual({
+        code: 404,
+        type: 'NotFound',
+        at: expect.any(Number),
+        message: 'Unable to find proposal',
+        url: '/v0.0/regtest/governance/proposals/123/votes?size=30&masternode=mine&cycle=0&all=false'
+      })
+    }
```

Comment:
This will pass even if the listGovProposalVotes promise resolves

Should do something like this instead:

```typescript
await expect(
  client.governance.listGovProposalVotes({ id: '123' })
).rejects.toThrowError(
  new WhaleApiException(...)
)
```